src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_params.py

```diff
@@ -435,6 +435,10 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
     with self.argument_context('storage share') as c:
         c.argument('share_name', share_name_type, options_list=('--name', '-n'))
 
+    with self.argument_context('storage share url') as c:
+        c.argument('unc', action='store_true', help='Output UNC network path.')
+        c.argument('protocol', arg_type=get_enum_type(['http', 'https'], 'https'), help='Protocol to use.')
```

Comment:
Will adding the parameter name in the call to get_enum_type e.g. "default='https'" make things clearer?

Reply:
Also do we want to unittest / scenario test protocol?

Reply:
Ideally, yes, however, it's infeasible to test every parameter for every command in larger modules like storage. 
As the `url` command makes no service calls and `protocol` is simply injected into a string, I don't think it is as useful to test.